### ISSUE TITLE
4 days of git-annex regression while testing ria_basics

### ISSUE THREAD

**Issue by yarikoptic** [datalad/datalad#4825]
For 4 days now daily tests of git annex build 
https://github.com/datalad/datalad-extensions/actions?query=workflow%3A%22Build+git-annex+snapshot%22
started to cancel (? not sure what that entails really) datalad test runs with

```
datalad.distributed.tests.test_ria_basics.test_gitannex_local ...   Cannot run git-annex-remote-!dne! -- It is not installed in PATH (/usr/lib/git-annex.linux/git-core:/usr/lib/git-annex.linux/bin:/usr/lib/git-annex.linux:/opt/hostedtoolcache/Python/3.7.8/x64/bin:/opt/hostedtoolcache/Python/3.7.8/x64:/home/linuxbrew/.linuxbrew/bin:/home/linuxbrew/.linuxbrew/sbin:/usr/share/rust/.cargo/bin:/home/runner/.config/composer/vendor/bin:/home/runner/.dotnet/tools:/snap/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/usr/lib/git-annex.linux/extra)
testremote store (generating test keys...) 
Remote Tests
  unavailable remote
##[error]The operation was canceled.
```

may be immediate ideas from @joeyh what could have lead to those new failures?

Previous instance related but closed since was no longer pertinent #4643 , at least it suggests that errors about `Cannot run git-annex-remote-!dne!` should be benign.

**Comment by kyleam** [comment]
I'm guessing the github action is getting canceled because it's hanging.  `test_gitannex_local` hangs for me locally.  The hang doesn't occur for me before the merge of the asyncexternal branch (95d9a3cf8), which lines up with the -extensions time frame.

I'll try to come up with a minimal reproducer.

**Comment by kyleam** [comment]
> I'll try to come up with a minimal reproducer.

Reported at https://git-annex.branchable.com/bugs/testremote_with_external_remote_hangs_after_asyncexternal_merge

**Comment by kyleam** [comment]
Joey pushed a fix: d00ce8241 (fix hang if external program is not available, 2020-08-19).

All the tests executed with `python -m nose -vs -x datalad.distributed.tests.test_ria_basics` now pass for me locally.  I'll keep an eye on the -extensions build and reopen this issue if the failure there isn't resolved.
